Tasks/Src/Task_VoltTempMonitor.c

```diff
@@ -47,15 +53,24 @@ void Task_VoltTempMonitor(void *p_arg) {
         bool charge_enable = true;
         CanMsg.id = VOLT_DATA;
         for (int i = 0; i < NUM_BATTERY_MODULES; i++){ //send all battery module voltage data
-            CanPayload.idx = i;
-            int voltage = Voltage_GetModuleMillivoltage(i);
-            CanData.w = voltage;
-            CanPayload.data = CanData;
-            CanMsg.payload = CanPayload;
-            CAN_Queue_Post(CanMsg);
+            
+            uint16_t voltage = Voltage_GetModuleMillivoltage(i);
+            voltage_averages[i] += voltage;
+
             if (voltage > CHARGE_DISABLE_VOLTAGE){
                 charge_enable = false;
             }
+
+            // Send voltages over CAN every ODR_VOLTAGE_AVERAGING iterations
+            if (task_cycle_counter % ODR_VOLTAGE_AVERAGING == 0) {
+                CanPayload.idx = i;
+                CanData.w = (int)(voltage_averages[i] / ODR_VOLTAGE_AVERAGING);
```

Comment:
could we just shift instead of divide since we're setting them to powers of 2. is the compiler smart enough to do that

Reply:
compiler is smart -- shift is a bit worse on readability